File: 75-sort-colors/sort-colors.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var sortColors = function(nums) {
    let start = 0, index = 0, end = nums.length - 1
    while(index <= end && start<end){
        if(nums[index] == 0){
            swap(nums, index, start)
            start++
            index++
        }
        else if(nums[index] == 2){
            swap(nums, index, end)
            end--
        }
        else index++
    }
    
};

function swap(arr, start, end){
    [arr[start], arr[end]] = [arr[end], arr[start]]
}

File: 75-sort-colors/sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        index, start, end = 0, 0, len(nums) - 1
        while start < end and index <= end: 
            if nums[index] == 0 : 
                nums[start], nums[index] = nums[index], nums[start]
                start+=1; index+=1
            elif nums[index] == 2 :
                nums[end], nums[index] = nums[index], nums[end]
                end-=1
            else : index+= 1
        